src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -703,8 +720,13 @@ abstract class RefChecks extends Transform {
                           // TODO: what is the optimal way to test for a raw type at this point?
                           // Compilation has already failed so we shouldn't have to worry overmuch
                           // about forcing types.
-                          if (underlying.isJavaDefined && pa.typeArgs.isEmpty && abstractSym.typeParams.nonEmpty)
-                            ". To implement a raw type, use %s[_]".format(pa)
+                          if (underlying.isJavaDefined && pa.typeArgs.isEmpty && abstractSym.typeParams.nonEmpty) {
+                            prepareRawTypeForMessage(abstractSym).fold(
+                              ". To implement a raw type, use %s[_]".format(pa)
```

Comment:
Yep, I think you can implement this message as `s". To implement this raw type, use ${rawToExistential(pa)}"` with a slight variation on a helper method it uses (patch below).

```diff
--- a/src/reflect/scala/reflect/internal/Types.scala
+++ b/src/reflect/scala/reflect/internal/Types.scala
@@ -3877,7 +3877,7 @@ trait Types
 
   def typeParamsToExistentials(clazz: Symbol, tparams: List[Symbol]): List[Symbol] = {
     val eparams = mapWithIndex(tparams)((tparam, i) =>
-      clazz.newExistential(newTypeName("?"+i), clazz.pos) setInfo tparam.info.bounds)
+      clazz.newExistential(tparam.name.toTypeName, clazz.pos) setInfo tparam.info.bounds)
 
     eparams map (_ substInfo (tparams, eparams))
   }
```

Would you be interested in pursuing this change and see if we can make this work for the whole compiler, or alternatively make a second `rawToExistentialPreservingNames`?